tests/app/test_model.py

```diff
@@ -657,3 +678,1616 @@ def test_serialize_service_filter_keys(
+
+    annual_billing = AnnualBilling(service=service, free_sms_fragment_limit=250000, financial_year_start=2022)
+    notify_db_session.add(annual_billing)
+    notify_db_session.commit()
+
+    result = annual_billing.serialize_free_sms_items()
+
+    assert result["free_sms_fragment_limit"] == 250000
+    assert result["financial_year_start"] == 2022
+
+
+def test_annual_billing_serialize_free_sms_items_only_returns_defined_fields(notify_db_session):
+    """Test to ensure we only get fields that are defined in the SerializedFreeSmsItems dataclass"""
+    service = create_service(service_name="Test Service", restricted=False)
+    notify_db_session.add(service)
+
+    annual_billing = AnnualBilling(
+        service=service,
+        free_sms_fragment_limit=250000,
+        financial_year_start=2022,
+    )
+    notify_db_session.add(annual_billing)
+    notify_db_session.commit()
+
+    result = annual_billing.serialize_free_sms_items()
+
+    # Should only contain fields defined in SerializedFreeSmsItems dataclass
+    assert set(result.keys()) == {"free_sms_fragment_limit", "financial_year_start"}
+
+
+def test_annual_billing_serialize(notify_db_session):
+    """Test the full serialization of annual billing"""
+    service = create_service(service_name="Test Service")
+    notify_db_session.add(service)
+
+    now = datetime.now(UTC)
+    annual_billing = AnnualBilling(
+        service=service,
+        free_sms_fragment_limit=250000,
+        financial_year_start=2022,
+        created_at=now,
+        updated_at=now + timedelta(days=1),
+    )
+    notify_db_session.add(annual_billing)
+    notify_db_session.commit()
+
+    result = annual_billing.serialize()
+
+    assert result["id"] == str(annual_billing.id)
+    assert result["free_sms_fragment_limit"] == 250000
+    assert result["service_id"] == str(service.id)
+    assert result["financial_year_start"] == 2022
+    assert result["created_at"] == now.strftime(DATETIME_FORMAT)
+    assert result["updated_at"] == (now + timedelta(days=1)).strftime(DATETIME_FORMAT)
+    assert result["service"] == {"id": str(service.id), "name": "Test Service"}
+
+
+def test_annual_billing_serialization_only_returns_defined_fields(notify_db_session):
+    """Test to ensure we only get fields that are defined in the serialization dataclass"""
+    service = create_service(service_name="Test Service")
+    notify_db_session.add(service)
+    annual_billing = AnnualBilling(
+        service=service,
+        free_sms_fragment_limit=250000,
+        financial_year_start=2022,
+    )
+    notify_db_session.add(annual_billing)
+    notify_db_session.commit()
+    serialized = annual_billing.serialize()
+
+    # Should only contain fields defined in SerializedAnnualBilling dataclass
+    assert set(serialized.keys()) == {
+        "id",
+        "free_sms_fragment_limit",
+        "service_id",
+        "financial_year_start",
+        "created_at",
+        "updated_at",
+        "service",
+    }
```

Comment:
Again, this could just be one test if you assert on the entire dictionary, not the keys and the contents separately

Reply:
Done